### DISCUSSION TITLE
How to deal with unhandled exception error

### DISCUSSION THREAD

**Discussion by Bittttter** [mobxjs/mobx-state-tree#1827] [Q&A]
Good Night!
When I build a large domain model, everything looks good.
No console errors, no ts errors.
But my fields are not updated as expected when initialized, something beyond expected error must have happened here, I mean, does MST have any api as like `unhandledrejection` that catches errors globally ? If not, how should I do best practice? Thanks for your help!

**Comment by kayvaninvemo** [comment]
Your question is a bit vague for me, can you provide some code snippet to demonstrate the issue, I do not understand what does this mean "my fields are not updated as expected when initialized", 
- How do you do "initialized"?
- What is the "expected" update?

In general MST throws error if it cannot create the model from the provided Snapshot:
```
  try{
    state = MyModel.create(initializationSnapshot);
  }catch(err){
    console.error(err);
  }
```

**Reply by Bittttter** [reply]
Sorry, I didn't describe the problem clearly.
This problem is because of `production mode` that made me never discover console error.😂
What does I raw thought before is add a global error catcher in MST. And I find the same issue in #933 .

Hope this is helpful...
